File: examples/governance_static_go/domain/stakeholder.go
package domain

import (
	"time"
	"github.com/google/uuid"
	"fmt"
)

type Stakeholder struct {
	ID string `json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Name string `json:"name"`
	Email string `json:"email"`
	Role string `json:"role"`
	Team string `json:"team"`
	Status string `json:"status"`
}

func NewStakeholder(name string, email string, role string, team string, status string) *Stakeholder {
	a := &Stakeholder{
		ID:        uuid.New().String(),
		Name: name,
		Email: email,
		Role: role,
		Team: team,
		Status: status,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
	}
	return a
}

func (a *Stakeholder) Validate() error {
	if a.Name == "" {
		return &ValidationError{Field: "name", Message: "name can't be blank"}
	}
	if a.Email == "" {
		return &ValidationError{Field: "email", Message: "email can't be blank"}
	}
	if a.Role != "" {
		validRole := map[string]bool{"assessor": true, "reviewer": true, "governance_board": true, "data_steward": true, "incident_reporter": true, "admin": true, "auditor": true}
		if !validRole[a.Role] {
			return &ValidationError{Field: "role", Message: fmt.Sprintf("role must be one of: assessor, reviewer, governance_board, data_steward, incident_reporter, admin, auditor, got: %s", a.Role)}
		}
	}
	return nil
}

File: examples/governance/identity_domain/lib/identity_domain/stakeholder/stakeholder.rb
require 'hecks/mixins/model'

module IdentityDomain
  class Stakeholder
    autoload :Lifecycle, "identity_domain/stakeholder/lifecycle"

    include Hecks::Model

    attribute :name
    attribute :email
    attribute :role
    attribute :team
    attribute :status

    # State predicates — see lifecycle.rb for full state machine
    def active?; status == "active"; end
    def deactivated?; status == "deactivated"; end

    VALID_ROLE = ["assessor", "reviewer", "governance_board", "data_steward", "incident_reporter", "admin", "auditor"].freeze unless defined?(VALID_ROLE)

    private

    def validate!
      raise ValidationError.new("name can't be blank", field: :name, rule: :presence) if name.nil? || (name.respond_to?(:empty?) && name.empty?)
      raise ValidationError.new("email can't be blank", field: :email, rule: :presence) if email.nil? || (email.respond_to?(:empty?) && email.empty?)
      if role && !VALID_ROLE.include?(role)
        raise ValidationError, "role must be one of: #{VALID_ROLE.join(', ')}, got: #{role}"
      end
    end
  end
end
